service/src/main/java/com/codeforcommunity/processor/MapProcessorImpl.java

```diff
@@ -122,4 +140,109 @@ public NeighborhoodGeoResponse getNeighborhoodGeoJson() {
             .collect(Collectors.toList());
     return new NeighborhoodGeoResponse(features);
   }
+
+  private <T> List<T> mergeSorted(List<T> l1, List<T> l2, Comparator<T> comparator) {
+    if (l1.isEmpty()) {
+      return l2;
+    }
+    if (l2.isEmpty()) {
+      return l1;
+    }
+    T element1 = l1.get(0);
+    T element2 = l2.get(0);
+    List<T> newList = new ArrayList<>();
+    if (comparator.compare(element1, element2) < 0) {
+      newList.add(element1);
+      newList.addAll(mergeSorted(l1.subList(1, l1.size()), l2, comparator));
+      return newList;
+    }
+    newList.add(element2);
+    newList.addAll(mergeSorted(l1, l2.subList(1, l2.size()), comparator));
+    return newList;
+  }
+
+  @Override
+  public SiteGeoResponse getSiteGeoJson() {
+    List<
+            Record9<
+                Integer,
+                Boolean,
+                Double,
+                String,
+                Timestamp,
+                String,
+                String,
+                BigDecimal,
+                BigDecimal>>
+        nonNullUserRecords =
+            this.db
+                .select(
+                    SITE_ENTRIES.ID,
+                    SITE_ENTRIES.TREE_PRESENT,
+                    SITE_ENTRIES.DIAMETER,
+                    SITE_ENTRIES.SPECIES,
+                    SITE_ENTRIES.UPDATED_AT,
+                    USERS.USERNAME,
+                    SITES.ADDRESS,
+                    SITES.LAT,
+                    SITES.LNG)
+                .from(SITE_ENTRIES)
+                .leftJoin(USERS)
```

Comment:
This may be me forgetting SQL but won't this include sites that have null users? We would want an innerJoin so that this list only has sites with a user right?

Reply:
I used a WHERE site_entries.user_id != null clause at the end, but I guess an inner join might be cleaner in this case.

Reply:
Inner join is definitely the cleaner approach here